zproject/backends.py

```diff
@@ -1676,9 +1676,10 @@ def social_auth_sync_user_attributes(
                 continue
             new_role = UserProfile.ROLE_API_NAME_TO_ID[attr_value]
         elif field_name.startswith("custom__"):
-            custom_profile_field_name_to_value[field_name.removeprefix("custom__")] = (
-                extra_attrs.get(attr_name)
-            )
+            attr_value = extra_attrs.get(attr_name)
+            if attr_value is None:
+                continue
+            custom_profile_field_name_to_value[field_name.removeprefix("custom__")] = attr_value
```

Comment:
There's some considerations about potentially different behavior being good here, copied from the commit message to just flag this here:

-------
We could consider a slightly different behavior here instead - when an
attribute is sent with no value in the SAMLResponse, that means the attr
has no value in the IdP's user directory - so perhaps a better behavior
would be to also remove the custom profile field value in Zulip. However
there are two issues with that:

1. It's not necessarily the best behavior, because an organization might
want the "user doesn't have this attribute set at the IdP level" state
to just mean that the user should be free to set the value manually in
Zulip if they wish. And having that value get reset on every login would
then be an issue. The implementation in this commit is consistent with
this philosophy.

2. There's some implementation difficulty - upstream
`self.get_attr(...)`, which we use for reading the attr value from the
SAMLResponse, doesn't distinguish between an attribute being sent with
no value and the attribute not being sent at all - in both cases it
returns None. So we'd need some extra work here with parsing the
SAMLResponse properly, to be able to know when the custom profile field
should get cleared.
-----

Reply:
Yeah, I think this is reasonable.